peachjam/adapters/adapters.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from tempfile import NamedTemporaryFile
+
+import magic
+import requests
+from dateutil import parser
+from django.core.files import File
+
+from peachjam.plugins import plugins
+
+logger = logging.getLogger(__name__)
+
+
+class Adapter:
+    def __init__(self, settings):
+        self.client = requests.session()
+        self.client.headers.update(
+            {
+                "Authorization": f"Token {settings['token']}",
+            }
+        )
+        self.url = settings["url"]
+
+    def check_for_updates(self, last_refreshed):
+        """Checks for documents updated since last_refreshed (which may be None), and returns a list
+        of document identifiers (expression FRBR URIs) which must be updated.
+        """
+        raise NotImplementedError()
+
+    def update_document(self, document_id):
+        """Update the document identified by some opaque id, returned by check_for_updates."""
+        raise NotImplementedError()
+
+    @classmethod
+    def name(cls):
+        return cls.__name__
+
+
+@plugins.register("ingestor-adapter")
+class IndigoAdapter(Adapter):

```

Comment:
Setting up the `self.client` object based on the settings must happen here, as part of the constructor, after the superclass's constructor has been called.

Reply:
Changed